window_doc.py

```diff
@@ -0,0 +1,59 @@
+from typing import Union
+
+from talon import Context, Module, actions, ui, app
+
+import os
+
+ctx = Context()
+mod = Module()
+
+@mod.action_class
+class Actions:
+
+    def file_manager_current_path_or_doc():
+        """Returns the current path being viewed/edited, first trying `file_manager_current_path` (in case it's implemented more
+        specifically), then falling back to `window.doc`.
+        """
+
+        path = actions.user.file_manager_current_path()
+        if not path:
+            return ui.active_window().doc.replace("%20", " ")
+
+    def validate_document(doc: str):
+        """Returns whether the given document path is valid, showing alerts if not."""
+
+        if not doc:
+            app.notify("No document to open", "The current application doesn't expose its document information")
```

Comment:
Probably 'window' rather than 'application' — for example the Office apps will let you reveal if local but not if the document is being opened over the Internet.